src/main/java/com/puppycrawl/tools/checkstyle/api/AbstractCheck.java

```diff
@@ -42,11 +43,8 @@
     /** The tokens the check is interested in. */
     private final Set<String> tokens = new HashSet<>();
 
-    /** The sorted set for collecting messages. */
-    private final SortedSet<LocalizedMessage> messages = new TreeSet<>();
-
-    /** The current file contents. */
-    private FileContents fileContents;
+    /** The check context. */
+    private final CheckContextHolder context = new CheckContextHolder();
```

Comment:
Why isn't this `ThreadLocal` instead of a field inside `CheckContextHolder`? Wouldn't that remove the need for a wrapper like `CheckContextHolder` and we could use `Context` directly?